### PR TITLE
TRMM, remove deprecated inplace trmm to favor outofplace/inplace trmm API

### PR DESCRIPTION
**Pull request by amcamd**
- this PR does 3 things
  - removes the macro HIPBLAS_V1 and code used for deprecating trmm_inplace
  - copies from daineAMD PR #504 to switch from trmm_inplace to trmm_outofplace. This contains most of the changes
  - call rocblas_Xtrmm in place of rocblas_Xtrmm_outofplace. This needs the changes in rocBLAS-internal PR # 1926
- This PR will fail tests untill rocBLAS-internal PR # 1926 is merged

### CODE REVIEW

File: clients/samples/example_strmm.cpp [hunk 1]
```diff
@@ -1,264 +0,0 @@
-/* ************************************************************************
- * Copyright (C) 2016-2023 Advanced Micro Devices, Inc. All rights reserved.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (TorreZuk) [inline clients/samples/example_strmm.cpp:4]: Why wasn't this just converted to show the new API use?

File: clients/include/hipblas.hpp [hunk 2]
```diff
@@ -25,10 +25,6 @@
 #ifndef _HIPBLAS_HPP_
 #define _HIPBLAS_HPP_
 
-#ifndef HIPBLAS_NO_DEPRECATED_WARNINGS
-#define HIPBLAS_NO_DEPRECATED_WARNINGS
-#endif
```
Comment (TorreZuk) [inline clients/include/hipblas.hpp:30]: Maybe just comment out for next deprecation use?

**Review by TorreZuk — APPROVED** [review]
LGTM but @daineAMD should review as had linked PR.

**Review by daineAMD — APPROVED** [review]

File: clients/samples/example_strmm.cpp (hunk 1 above)
Comment (amcamd) [reply clients/samples/example_strmm.cpp]: Thanks for catching this. Added back example_strmm.cpp in next commit.

File: docs/deprecation.rst [hunk 3]
```diff
@@ -90,3 +90,49 @@ Packed int8x4 was removed as support for arbitrary dimensioned int8_t data is a
 * enum hipblasInt8Datatype_t was removed
 * function hipblasSetInt8Datatype was removed
 * function hipblasGetInt8Datatype was removed
+
+Legacy BLAS in-place trmm
```
[the remark this answers is not in the document]
Comment (amcamd) [reply docs/deprecation.rst:97]: Thanks for finding this, added `Removed in hipBLAS 2.0` section in next commit.

File: library/include/hipblas.h [hunk 4]
```diff
@@ -14561,178 +14561,19 @@ HIPBLAS_EXPORT hipblasStatus_t hipblasZhemmStridedBatched(hipblasHandle_t
...
-        lower triangular part of the array  A must contain the lower
-        triangular matrix  and the strictly upper triangular part of
-        A is not referenced. Here k is m when  side == rocblas_side_left
-        and is n when side == rocblas_side_right.
+        op( A ) = A   or   op( A ) = A^T   or   op( A ) = A^H.
 
-        Note that when  diag == rocblas_diagonal_unit  the diagonal elements of
-        A  are not referenced either,  but are assumed to be  unity.
+    Note that trmm can provide in-place functionality by passing in the same address for both
+    matrices B and C.
```
[the remark this answers is not in the document]
Comment (amcamd) [reply library/include/hipblas.h]: Thanks for finding this, added `requires ldb == ldc` in next commit.

**Pull request merged**